### PR TITLE
Added modal functions 

### PR DESCRIPTION
**Pull request by matodorovic**
### Description
This PR adds functions to booking modal. 

### Screenshots
Skärmavbild 2022-12-01 kl  16 18 44

Skärmavbild 2022-12-01 kl  16 23 14

Skärmavbild 2022-12-01 kl  16 39 54

Skärmavbild 2022-12-01 kl  16 40 04

Skärmavbild 2022-12-01 kl  16 29 44

### Changes

- Adds checking if date input is not empty
- Adds function loadTimes to fetch times from api
- Adds checking if name, email, participants or time is not empty
- Adds text options to select participants and select time
- Adds function reserveTime to send booking to api

### CODE REVIEW

**Review by solidtop — APPROVED** [review]
Det ser bra ut! Både kod och visuellt på modalen :) Good work!

**Review by davidlidebrandt — APPROVED** [review]
Det ser jättebra ut!

File: src/js/modal.js [hunk 1]
```diff
@@ -1,156 +1,190 @@
...
+    selectParticipants.appendChild(opt);
     form.appendChild(selectParticipants);
 
     let amountOfOptions = challenge.maxParticipants - challenge.minParticipants;
     for (let i = 0; i < amountOfOptions; i++) {
-        const option = document.createElement('option');
+        const option = document.createElement("option");
         option.textContent = `${challenge.minParticipants + i} participants`;
         selectParticipants.appendChild(option);
     }
```
Comment (JonathanHolmgren) [inline src/js/modal.js:131]: Denna kod kommer generera minus en deltagarna från vad som kommer ifrån API:n. 
För att lösa detta kan man göra en quick fix och bara plussa på 1.

let amountOfOptions = challenge.maxParticipants - challenge.minParticipants + 1;

**Review by JonathanHolmgren — CHANGES_REQUESTED** [review]
Koden ser bra ut, hade bara en liten detalj som gällde antalet select participants.

File: src/js/modal.js (hunk 1 above)
Comment (matodorovic) [reply src/js/modal.js:131]: I for loopen changed from i < amountOfParticipants to <= so it solves it.

**Review by JonathanHolmgren — APPROVED** [review]

File: src/script.js [hunk 2]
```diff
@@ -1,11 +1,21 @@
...
 
 (async () => {
-  let challangesArray = await loadData();
+  let challengesArray = await loadData();
 
-  document.getElementById("text-filter").addEventListener("keyup", () => {
-    const newArray = filterText(challangesArray);
-    console.log(newArray);
+  document.getElementById('text-filter').addEventListener('keyup', () => {
+     return filterText(challengesArray);
```
Comment (richardolsson) [inline src/script.js:10]: Return values are irrelevant for event handlers. Returning something from an event handler does nothing.

File: src/js/modal.js [hunk 3]
```diff
@@ -1,156 +1,190 @@
...
     div.appendChild(message);
 
-    const button = document.createElement('button');
-    button.classList.add('button', 'secondary');
-    button.textContent = 'Back to challenges';
-    button.addEventListener('click', removeModal);
+    const button = document.createElement("button");
+    button.classList.add("button", "secondary");
+    button.textContent = "Back to challenges";
+    button.addEventListener("click", removeModal);
```
Comment (richardolsson) [inline src/js/modal.js:187]: This doesn't actually change any logic, just the format of the code (switching between `"` and `'` quote marks).

Changes like these make a diff (the visualization of changes in a commit or PR) hard to follow. Avoid meaningless changes like these by agreeing on a format and sticking to it.

File: src/js/modal.js [hunk 4]
```diff
@@ -1,156 +1,190 @@
...
     });
 
     return form;
 }
 
-function step2(challenge, times) {
-
-    const form = document.createElement('form');
-    form.classList.add('modal-content');
+async function step2(challenge, times) {
```
Comment (richardolsson) [inline src/js/modal.js:68]: I see no reason for this function to be async. It only means that any code that uses this function will now also need to be async, and any code using _that_ code will also need to be async, and so on until it contaminates everything.

Avoid making functions async if they don't actually do async work.

File: src/js/filterText.js [hunk 5]
```diff
@@ -1,10 +1,10 @@
 export default function filterText(data) {
   let inputvalue = document.getElementById("text-filter").value;
-  //console.log(inputvalue);
-
+  console.log(inputvalue);
+  if (inputvalue) {
```
Comment (richardolsson) [inline src/js/filterText.js:4]: I would expect this change to include indenting the next four lines, but it doesn't, resulting in inconsistent indentation.

**Review by richardolsson — COMMENTED** [review]
Good work! I've added some feedback below. Let me know if you have any questions!
